leaf-ui/js/object/Class.js

```diff
@@ -338,12 +346,26 @@ class AnalysisResult {
         this.timePointPathSize;
         this.elementList; 
         this.allSolution;
-        this.elementListPercentEvals;
         this.isPathSim = false; //used for slider visualization
         this.colorVis; //color visualization for analysis mode  
         this.selectedTimePoint; //find where slider is initialized and set timepoint in here. Also place it in update function
-        this.maxTimePoint;
+        this.timeScale;
+        
+
+
+
+
+
     }
+    getTimeScale() {
+        this.timeScale = Number(this.timePointPath.length) - 1;
+        return this;
```

Comment:
You shouldn't have to `return this` -- that would return the AnalysisResult object you are already working on, which won't be necessary